core/cli/src/lib.rs

```diff
@@ -307,18 +307,33 @@ pub struct ParseAndPrepareBuildSpec<'a> {
 
 impl<'a> ParseAndPrepareBuildSpec<'a> {
 	/// Runs the command and build the chain specs.
-	pub fn run<G, S, E>(
+	pub fn run<C,G, S, E>(
 		self,
 		spec_factory: S
 	) -> error::Result<()> where
 		S: FnOnce(&str) -> Result<Option<ChainSpec<G, E>>, String>,
+		C: Default,
 		G: RuntimeGenesis,
 		E: ChainSpecExtension,
 	{
 		info!("Building chain spec");
 		let raw_output = self.params.raw;
 		let mut spec = load_spec(&self.params.shared_params, spec_factory)?;
-		with_default_boot_node(&mut spec, self.params, self.version)?;
+
+		if spec.boot_nodes().is_empty() {
```

Comment:
```suggestion
		if spec.boot_nodes().is_empty() && !self.params.disable_default_bootnode {
```

Why did you remove this?

Reply:
not intentionally, this was moved before your patch came in and I forgot it in the merge. Good catch!